kong/plugins/datadog/handler.lua

```diff
@@ -98,6 +98,18 @@ function DatadogHandler:log(conf)
     return
   end
 
+  if conf.env_prefix then
+    local env_host = os.getenv(conf.env_prefix .. '_HOST')
+    local env_port = tonumber(os.getenv(conf.env_prefix .. '_PORT'))
+    if not env_host or not env_port then
+      kong.log.err("both " .. conf.env_prefix .. "_HOST and " .. conf.env_prefix .. "_PORT must be set ",
+                   "when config.env_prefix is configured")
+      return
+    end
+    conf.host = env_host
```

Comment:
this is not ideal, we are overriding it per request and implictly, potential improvments are:
- use custom_validator to do this in schema part (is the fields mutable in custom_validator?)
- use a cache to memoize env vars, since usually those don't change once process are started

Reply:
another issue is that nginx removes environment variables at some point (worker fork?), so i don't think they're available at request time.  what about loading a global dict of environment at during startup?

Reply:
@javierguerragiraldez agreed, imo we should not use variable env variables. This becomes extremely messy, if you run many dataplanes, multiple of those variables must be defined on all of the dataplanes then. Big can of worms imo.
So keep them hardcoded, as long as they are properly namespaced that should not be an issue. 

that said, we should drop this and use https://github.com/Kong/kong/pull/7632 instead. My 2cts.